### ISSUE TITLE
PluginCliTests fail on windows

### ISSUE THREAD

**Issue by nik9000** [elastic/elasticsearch#58605]
**Build scan**:
[scan](https://gradle-enterprise.elastic.co/s/u7e46d62t4hfy/tests/:qa:os:destructiveDistroTest.default-windows-archive-no-jdk/org.elasticsearch.packaging.test.PluginCliTests/test20SymlinkPluginsDir#1)

**Repro line**: 

**Reproduces locally?**:

**Applicable branches**: master and 7.x

**Failure history**:

**Failure excerpt**:
```
uncaught exception in thread [main]
	at java.security.AccessControlContext.checkPermission(AccessControlContext.java:472)
	at java.security.AccessController.checkPermission(AccessController.java:886)
java.lang.IllegalStateException: failed to load plugin custom-settings due to jar hell
Likely root cause: java.security.AccessControlException: access denied ("java.io.FilePermission" "C:\tmp\symlinked-plugins3650558477987461918\custom-settings\custom-settings-7.9.0-SNAPSHOT.jar" "read")
	at java.util.zip.ZipFile.<init>(ZipFile.java:216)
	at java.lang.SecurityManager.checkRead(SecurityManager.java:888)
	at java.util.zip.ZipFile.<init>(ZipFile.java:155)
	at java.util.jar.JarFile.<init>(JarFile.java:103)
	at java.lang.SecurityManager.checkPermission(SecurityManager.java:549)
```

**Comment by rjernst** [comment]
I was able to reproduce this by using java 8 SYSTEM_JAVA_HOME with the windows no-jdk archive. It appears java 8 has included some behavior which went into java 9 regarding canonical paths. I'm backporting the change we already have in master to always use the more extensive permissions we add for java9+: https://github.com/elastic/elasticsearch/pull/58628.

**Comment by rjernst** [comment]
Closed by #58628
